src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/HashPartitioningItemMapper.java

```diff
@@ -110,60 +96,24 @@ private void addPhysicalKeys(Map<String, AttributeValue> virtualItem, Map<String
             }
             if (virtualHkValue != null && virtualRkValue != null) {
                 // add physical HK
-                putPhysicalHashKey(physicalItem, virtualPK.getHashKeyType(), virtualHkValue, physicalPK);
+                physicalItem.put(physicalPK.getHashKey(), keyMapper.toPhysicalHashKey(virtualHkType, virtualHkValue));
 
                 // add physical RK
-                ByteBuffer physicalRangeKeyValue = AttributeBytesConverter.toBytes(
-                    virtualPK.getHashKeyType(), virtualHkValue, virtualPK.getRangeKeyType().get(), virtualRkValue);
-                putBytesInField(physicalItem, physicalPK.getRangeKey().get(), physicalRangeKeyValue);
-
-                // remove virtual HK and RK if needed
-                physicalItem.remove(virtualPK.getHashKey());
-                physicalItem.remove(virtualPK.getRangeKey().get());
+                physicalItem.put(physicalPK.getRangeKey().get(),
+                    keyMapper.toPhysicalRangeKey(virtualPK, virtualHkValue, virtualRkValue));
```

Comment:
We now maintain all the original virtual PK/index field values, which makes our lives a lot easier, since we don't have to translate filter & projection expressions.